### PR TITLE
Implement updated name_spec for displayForm.

### LINKED ISSUE #479: Implement updated name_spec for displayForm

### PR DESCRIPTION
**Pull request by justinlittman**
closes #479

**NOTE:  Changes to openapi.yml require updating openapi.yml for sdr-api and dor-services-app and generating models - see README.**

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: lib/cocina/models/builders/name_title_group_builder.rb [hunk 1]
```diff
@@ -92,7 +92,6 @@ def self.value_slices(desc_value)
           # ignoring groupedValue
           slices.flatten
         end
-        # private_class_method :value_slices
```
Comment (ndushay) [inline lib/cocina/models/builders/name_title_group_builder.rb:95]: thank you!

File: lib/cocina/models/mapping/from_mods/name_builder.rb [hunk 2]
```diff
@@ -116,23 +126,68 @@ def build_name_parts(name_node)
...
-              vals = name_part_nodes.filter_map do |name_part|
-                build_name_part(name_node, name_part, default_type: name_node['type'] != 'corporate').presence
-              end
-              parts << { structuredValue: vals }.merge(common_authority(name_node)).merge(common_lang_script(name_node))
+              name = build_structured_value_name(name_node, name_part_nodes)
+              parts << name.merge(common_authority(name_node)).merge(common_lang_script(name_node))
+              # vals = name_part_nodes.filter_map do |name_part|
+              #   build_name_part(name_node, name_part, default_type: name_node['type'] != 'corporate').presence
+              # end
+              # parts << { structuredValue: vals }.merge(common_authority(name_node)).merge(common_lang_script(name_node))
```
Comment (ndushay) [inline lib/cocina/models/mapping/from_mods/name_builder.rb:137]: cruft?

**Review by ndushay — APPROVED** [review]
I was going to complement you on your method names and refactoring, but it turns out a lot of that was from my attempt.  Haha.  No wonder I liked it.

File: lib/cocina/models/mapping/from_mods/name_builder.rb (hunk 2 above)
Comment (justinlittman) [reply lib/cocina/models/mapping/from_mods/name_builder.rb]: Yup. Removed.

**Pull request merged**